src/v/kafka/server/materialized_partition.h

```diff
@@ -52,6 +54,9 @@ class materialized_partition final : public kafka::partition_proxy::impl {
           std::vector<cluster::rm_stm::tx_range>());
     }
 
+    /// A materialized partition is a leader if its source topic is
+    bool is_leader() const override { return _partition->is_leader(); }
```

Comment:
I would move adding `is_leader()` method to separate commit.

Reply:
Only the is_leader method? This commit adds another argument to the constructor of `materialized_partition` and edits the `make_materialized_partition` method, to me it sounds like a solid connected piece

Reply:
Going to resolve this in favor of Noahs suggestion to break out this PR